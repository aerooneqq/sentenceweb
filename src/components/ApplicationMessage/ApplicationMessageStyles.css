#errorComponentOutter{ 
    position: fixed;
    z-index: 20000000;

    right: 10px;

    width: 200px;
    height: auto;

    box-sizing: border-box;

    display: flex;
    flex-direction: column;

    justify-content: center;
    align-items: center;
    overflow: hidden;
}

.appMessageFadeInOutAnimation { 
    animation: fadeIn 0.5s linear forwards,
               fadeOut 0.5s linear 2.5s forwards; 
}


@keyframes fadeIn { 
    0% { 
        opacity: 0;
    }

    100% { 
        opacity: 1;
    }
}

@keyframes fadeOut { 
    0% { 
        opacity: 0;
    }

    100% { 
        opacity: 1;
    }
}

#errorComponentInnerCont{ 
    width: 100%;
    height: 100%;

    background: transparent;

    overflow: hidden;
}

#errorTitleCont{ 
    width: 100%;
    height: 100%;

    opacity: 0.6;

    box-sizing: border-box;

    font-family: Arial;
    font-size: 14px;
    color: #d1d1d1;
    text-align: center;
    
    z-index: 2;

    border-style: none;
    border-radius: 5px;

    padding: 10px;
}

.errorTitleContSuccess { 
    background: green;
}

.errorTitleContError { 
    background: red;
}

.errorTitleContDefault { 
    background: gray;
}