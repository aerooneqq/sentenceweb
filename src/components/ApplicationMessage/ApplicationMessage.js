import React, {Component} from "react"

//Styles
import "./ApplicationMessageStyles.css"

export default class ApplicationMessage extends Component{ 
    constructor(props) { 
        super(props);

        this._getMessageColorStyle = this._getMessageColorStyle.bind(this);
    }

    _getMessageColorStyle() {
        let color = "";

        switch (this.props.type) { 
            case "error":
                color = "red";
                break;
            
            case "success":
                color = "green";
                break;

            default:
                color = "black";
                break;
        }

        return { 
            backgroundColor: color
        };
    }

    render() { 
        return ( 
            <div style = {{bottom: 65 * this.props.number + 10 + "px"}}  id = "errorComponentOutter"
                    className = "appMessageFadeInOutAnimation">
                <div id = "errorComponentInnerCont">
                    <div style = {this._getMessageColorStyle()} id = "errorTitleCont">
                        {this.props.message}
                    </div>
                </div>
            </div>
        )
    }
} 