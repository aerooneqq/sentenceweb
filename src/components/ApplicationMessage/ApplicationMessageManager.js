import React from "react"
import ApplicationMessage from "./ApplicationMessage";

var alertAppMessageFunc = null;
var messages = []

function alertAppMessage(message, type) { 
    messages.push(<ApplicationMessage message = {message} type = {type} number = {messages.length} />);

    alertAppMessageFunc(messages)
    setTimeout(() => { 
        if (messages.length > 0 ) { 
            messages.pop();
            alertAppMessageFunc(messages);
        }
    }, 2500);
}

function setAppMesageParams(alertFunc) { 
    alertAppMessageFunc = alertFunc;
}

export {alertAppMessage, setAppMesageParams}
