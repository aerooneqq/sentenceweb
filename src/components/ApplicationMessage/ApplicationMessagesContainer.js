import React, {Component} from "react"; 

import {setAppMesageParams} from "./ApplicationMessageManager";

export default class ApplicationMessagesContainer extends Component { 
    constructor(props) { 
        super(props);

        this.state = { 
            messages: []
        }

        this.alertAppMessages = this.alertAppMessages.bind(this);

        setAppMesageParams(this.alertAppMessages);
    }

    alertAppMessages(messages) { 
        this.setState(() => { 
          return { 
            messages: messages
          }
        });
      }

    render() { 
        return ( 
            <div>
                {this.state.messages}
            </div>
        )
    }
}