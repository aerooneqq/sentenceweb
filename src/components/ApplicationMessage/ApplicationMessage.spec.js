import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import ApplicationMessage from './ApplicationMessage';

describe('application message', () => { 
    it("should render correctly", () => { 
        const props = { 
            type: "success",
            message: "Message"
        }

        const applicationMessage = shallow(<ApplicationMessage {...props} />);
        expect(shallowToJson(applicationMessage)).toMatchSnapshot();
    });

    it('should display message', () => { 
        const props = { 
            type: "success",
            message: "The message"
        }

        const applicationMessage = shallow(<ApplicationMessage {...props} />);
        expect(applicationMessage.find("#errorTitleCont").text()).toBe("The message");
    });

    it ("should have success color class when success", () => { 
        const props = { 
            type: "success",
            message: "Message"
        }

        const applicationMessage = shallow(<ApplicationMessage {...props} />);
        expect(applicationMessage.find("#errorTitleCont").text()).toBe("Message");
        expect(applicationMessage.find("#errorTitleCont").hasClass("errorTitleContSuccess"));
    });

    it("must have error color class when error", () => { 
        const props = { 
            type: "error",
            message: "Yo!"
        }

        const applicationMessage = shallow(<ApplicationMessage {...props}/>);
        expect(applicationMessage.find("#errorTitleCont").text()).toBe("Yo!");
        expect(applicationMessage.find("#errorTitleCont").hasClass("errorTitleContError"));
    });

    it("must have default color class when type is not specified", () => { 
        const props = { 
            message: "Hello!"
        };

        const applicationMessage = shallow(<ApplicationMessage {...props} />);
        expect(applicationMessage.find("#errorTitleCont").text()).toBe("Hello!");
        expect(applicationMessage.find("#errorTitleCont").hasClass("defaultTitleContError"));
    });

    it("must render correctly when props are null", () => { 
        const props = { 
            message: null,
            type: null
        }

        const applicationMessage = shallow(<ApplicationMessage {...props} />);
        expect(applicationMessage.find("#errorTitleCont").text()).toBe("");
        expect(applicationMessage.find("#errorTitleCont").hasClass("defaultTitleContError"));
    });
});