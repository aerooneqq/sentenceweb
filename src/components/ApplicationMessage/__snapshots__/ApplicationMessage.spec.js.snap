// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application message should render corectly 1`] = `
<div
  className="appMessageFadeInOutAnimation"
  id="errorComponentOutter"
  style={
    Object {
      "bottom": "NaNpx",
    }
  }
>
  <div
    id="errorComponentInnerCont"
  >
    <div
      id="errorTitleCont"
      style={
        Object {
          "backgroundColor": "black",
        }
      }
    />
  </div>
</div>
`;

exports[`application message should render correctly 1`] = `
<div
  className="appMessageFadeInOutAnimation"
  id="errorComponentOutter"
  style={
    Object {
      "bottom": "NaNpx",
    }
  }
>
  <div
    id="errorComponentInnerCont"
  >
    <div
      class={[Function]}
      id="errorTitleCont"
    >
      Message
    </div>
  </div>
</div>
`;
